sensors/hall effect sensor/rpm/hall_effect_sensor.h

```diff
@@ -0,0 +1,46 @@
+//NOTE: CODE TAKEN FROM: https://www.youtube.com/watch?v=u2uJMJWsfsg
+
+#ifndef HALL_EFFECT_SENSOR_H
+#define HALL_EFFECT_SENSOR_H
+
+class HallEffectSensor : public Sensor
+{
+public:
+    HallEffectSensor(const int &pin);
+    void read_sensor_value();
+    double get_sensor_value();
```

Comment:
The base_sensor's Sensor class has a defined get_sensor_value() function, might be preferable for scalability. It returns m_sensor_value which you could use instead of RPM. That said, the code would become less readable, so I'll defer to the others which approach we want to take